libs/language-server/src/lib/validation/checks/property-assignment.spec.ts

```diff
@@ -0,0 +1,162 @@
+    );
+  }
+
+  beforeAll(() => {
+    // Register test extension
+    useExtension(new TestLangExtension());
+    // Create language services
+    const services = createJayveeServices(NodeFileSystem).Jayvee;
+    locator = services.workspace.AstNodeLocator;
+    // Parse function for Jayvee (without validation)
+    parse = parseHelper(services);
+  });
+
+  afterEach(() => {
+    // Reset mock
+    validationAcceptorMock.mockReset();
+  });
+
+  it('error on invalid property name', async () => {
+    const text = readJvTestAsset(
+      'property-assignment/invalid-unknown-property.jv',
+    );
+
+    await parseAndValidatePropertyAssignment(text);
+
+    expect(validationAcceptorMock).toHaveBeenCalledTimes(1);
+    expect(validationAcceptorMock).toHaveBeenCalledWith(
+      'error',
+      `Invalid property name "unknownProperty".`,
+      expect.any(Object),
+    );
+  });
+
+  describe('runtime parameter for property', () => {
+    it('should have no error on runtime parameter for text property', async () => {
+      const text = readJvTestAsset(
+        'property-assignment/valid-runtime-property.jv',
+      );
+
+      await parseAndValidatePropertyAssignment(text);
+
+      expect(validationAcceptorMock).toHaveBeenCalledTimes(0);
+    });
+
+    it('error on runtime parameter for regex property', async () => {
+      const text = readJvTestAsset(
+        'property-assignment/invalid-runtime-property.jv',
+      );
+
+      await parseAndValidatePropertyAssignment(text);
+
+      expect(validationAcceptorMock).toHaveBeenCalledTimes(1);
+      expect(validationAcceptorMock).toHaveBeenCalledWith(
+        'error',
+        `Runtime parameters are not allowed for properties of type regex`,
+        expect.any(Object),
+      );
+    });
+  });
+
+  it('error on invalid property typing', async () => {
+    const text = readJvTestAsset(
+      'property-assignment/invalid-property-type.jv',
+    );
+
+    await parseAndValidatePropertyAssignment(text);
+
+    expect(validationAcceptorMock).toHaveBeenCalledTimes(1);
+    expect(validationAcceptorMock).toHaveBeenCalledWith(
+      'error',
+      `The value needs to be of type text but is of type integer`,
+      expect.any(Object),
+    );
+  });
+
+  it('info on simplifiable property expression', async () => {
+    const text = readJvTestAsset('property-assignment/valid-simplify-info.jv');
+
+    await parseAndValidatePropertyAssignment(text);
+
+    expect(validationAcceptorMock).toHaveBeenCalledTimes(1);
+    expect(validationAcceptorMock).toHaveBeenCalledWith(
+      'info',
+      `The expression can be simplified to 1019`,
+      expect.any(Object),
+    );
+  });
+
+  it('info on non simplifiable property expression', async () => {
+    const text = readJvTestAsset(
+      'property-assignment/valid-uneccessarysimplify-info.jv',
+    );
+
+    await parseAndValidatePropertyAssignment(text);
+
+    expect(validationAcceptorMock).toHaveBeenCalledTimes(0);
+  });
```

Comment:
Could you add an additional test case to cover the recently added feature #352?

Reply:
That is desired behavior for now. So far, the language server is only capable of suggesting simplifications, where sub-expressions can be evaluated statically (i.e. when expressions don't contain free variables such as `input`). At the moment, the simplification engine has no assumptions on individual operators (e.g. it does not know that `floor(floor(x)) == x`).